lib/rack/attack/cache.rb

```diff
@@ -30,6 +30,18 @@ def write(unprefixed_key, value, expires_in)
         store.write("#{prefix}:#{unprefixed_key}", value, :expires_in => expires_in)
       end
 
+      def reset_count(unprefixed_key, period)
+        epoch_time = Time.now.to_i
+        # Add 1 to expires_in to avoid timing error: http://git.io/i1PHXA
+        expires_in = period - (epoch_time % period) + 1
+        key = "#{(epoch_time / period).to_i}:#{unprefixed_key}"
```

Comment:
@stanhu this nearly a dupe of lines 18-21. What do you think about extracting it to a private method?